plugins/micron/micron-nvme.c

```diff
@@ -303,8 +303,14 @@ static int SetupDebugDataDirectories(char *strSN, char *strFilePath,
 	}
 	strMainDirName[nIndex] = '\0';
 
+	if (setuid(getuid())) {
+		err = -1;
+		goto exit_status;
+	}
+
 	j = 1;
-	while (!stat(strMainDirName, &dirStat)) {
+	while (!(fd = open(strMainDirName, 0))) {
+		close(fd);
```

Comment:
This doesn't make any sense. It's the same. The problem is that we check if strMainDirName exists and then we do later a rmdir. This has to be tied together.

Reply:
Sorry the change not correct and should be fixed as below.
```diff
- 	while (!(fd = open(strMainDirName, 0))) {
+ 	while (!!(fd = open(strMainDirName, 0))) {
```
But changed to fix by unusing stat() and it can be checked by the mkdir errno instead if strMainDirName exists.
The code behavior itself same with the original code scanned the alert but the alert itself can be resolved by the changes. (The alert itself can be resolved as same by the previous changes also.)
Since seems the scanner just checks if using stat() and rmdir() only.
Also the issue not only for strMainDirName but seems same for strOSDirName and strCtrlDirName but just not detected by the scanner.

Reply:
Yes, using mkdir directly to check if the patch exists, is the canonical approach for this. Let's go with it.